### PR TITLE
Vote 2023-10-31

### PR DESCRIPTION
**Pull request by kovalgek**
#### I. stETH transfers to  RCC PML ATC
1. Transfer TBA stETH to RCC 0xDE06d17Db9295Fa8c4082D4f73Ff81592A3aC437
2. Transfer TBA stETH to PML 0x17F6b2C738a63a8D3A113a228cfd0b373244633D
3. Transfer TBA stETH to ATC 0x9B1cebF7616f2BC73b47D226f90b01a7c9F86956

#### II. Change the on-chain name of node operator with id 27 from 'Prysmatic Labs' to 'Prysm Team at Offchain Labs'
4. Grant NodeOperatorsRegistry.MANAGE_NODE_OPERATOR_ROLE to voting
5. Change node operator name from Prysmatic Labs to Prysm Team at Offchain Labs
6. Revoke MANAGE_NODE_OPERATOR_ROLE from Voting

**todo**:
- [x] update stETH amount before voting

The vote REJECTED.

### CODE REVIEW

**Comment by BATMAH69** [comment]
> @rkolpakov @BATMAH69 Hey folks, can you point out the new factories and briefly uncover the details about their connection?

I don't have information about it but @rkolpakov have

File: scripts/vote_2023_10_31.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    pml_stable_topup_factory = interface.TopUpAllowedRecipients("0x4E6D3A5023A38cE2C4c5456d3760357fD93A22cD")
+    atc_stable_topup_factory = interface.TopUpAllowedRecipients("0x67Fb97ABB9035E2e93A7e3761a0d0571c5d7CD07")
+
+    # todo: change addresses
+    rcc_stable_registry = interface.AllowedRecipientRegistry("0xDc1A0C7849150f466F07d48b38eAA6cE99079f80")
+    pml_stable_registry = interface.AllowedRecipientRegistry("0xDFfCD3BF14796a62a804c1B16F877Cf7120379dB")
+    atc_stable_registry = interface.AllowedRecipientRegistry("0xe07305F43B11F230EaA951002F6a55a16419B707")
+
+    call_script_items = [
+        # 1. Switch off TopUp ETs for RCC PML ACT
```
Comment (TheDZhon) [inline scripts/vote_2023_10_31.py:51]: There should be finance limits update:
https://github.com/lidofinance/scripts/blob/33034fe893a190991ee26e30c3118789feb5dedd/archive/scripts/vote_2022_12_06_1.py#L61-L102

see: https://discord.com/channels/761182643269795850/1163831729635733574/1166262759630061588
and https://github.com/lidofinance/lido-improvement-proposals/blob/develop/LIPS/lip-13.md

**Review by TheDZhon — CHANGES_REQUESTED** [review]
:eyes:

**Comment by TheDZhon** [comment]
Let's add new ABIs to the PR @kovalgek :pray:

**Comment by TheDZhon** [comment]
Fixed the acceptance and regression tests that were failing due to the NO-1 offboarding.

image

File: scripts/vote_2023_10_31.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+    call_script_items = [
+        # I. Add USDT and USDC to EVMScriptExecutor permissions
+        # 1. Revoke role CREATE_PAYMENTS_ROLE from EVM script executor
+        encode_permission_revoke(
+            target_app=contracts.finance,
+            permission_name="CREATE_PAYMENTS_ROLE",
+            revoke_from=EASYTRACK_EVMSCRIPT_EXECUTOR,
+        ),
+        # 2. Grant role CREATE_PAYMENTS_ROLE to EasyTrack EVMScriptExecutor 0xFE5986E06210aC1eCC1aDCafc0cc7f8D63B3F977
+        # with limits: 1000 ETH, 1000 stETH, 5M LDO, 2M DAI, 2M USTD, 2M USDC
```
Comment (zuzueeka) [inline scripts/vote_2023_10_31.py:179]: typo: USTD -> USDT

File: scripts/vote_2023_10_31.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+            permissions=create_permissions(contracts.finance, "newImmediatePayment")
+            + create_permissions(atc_stable_registry, "updateSpentAmount")[2:],
+        ),
+
+        # III. stETH transfers to RCC PML ATC
+        # 9. Transfer TBA stETH to RCC 0xDE06d17Db9295Fa8c4082D4f73Ff81592A3aC437
+        make_steth_payout(
+            target_address=rcc_multisig_address,
+            steth_in_wei=1 * (10**18),
+            reference="Fund Gas Funder multisig"
```
Comment (zuzueeka) [inline scripts/vote_2023_10_31.py:219]: Fund RCC multisig

File: scripts/vote_2023_10_31.py [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+        ),
+        # 10. Transfer TBA stETH to PML 0x17F6b2C738a63a8D3A113a228cfd0b373244633D
+        make_steth_payout(
+            target_address=pml_multisig_address,
+            steth_in_wei=1 * (10**18),
+            reference="Fund Gas Funder multisig"
```
Comment (zuzueeka) [inline scripts/vote_2023_10_31.py:225]: Fund PML multisig

File: scripts/vote_2023_10_31.py [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+        ),
+        # 11. Transfer TBA stETH to ATC 0x9B1cebF7616f2BC73b47D226f90b01a7c9F86956
+        make_steth_payout(
+            target_address=atc_multisig_address,
+            steth_in_wei=1 * (10**18),
+            reference="Fund Gas Funder multisig"
```
Comment (zuzueeka) [inline scripts/vote_2023_10_31.py:231]: Fund ATC multisig

File: tests/test_2023_10_31.py [hunk 6]
```diff
@@ -0,0 +1,657 @@
...
+        contracts.easy_track,
+        atc_trusted_caller_and_recepient,
+        atc_stable_topup_factory,
+        contracts.dai_token,
+        [atc_trusted_caller_and_recepient],
+        [10 * 10**18],
+        stranger,
+    )
+    check_add_and_remove_recipient_with_voting(atc_stable_registry, helpers, LDO_HOLDER_ADDRESS_FOR_TESTS, contracts.voting)
+
```
Comment (zuzueeka) [inline tests/test_2023_10_31.py:510]: Could you please add tests for stETH payments?
1. agent’s balance has decreased by the expected number. 
2. multisig balances increased by expected amounts.
3. agent delta = sum of multisig deltas.

File: tests/test_2023_10_31.py [hunk 7]
```diff
@@ -0,0 +1,657 @@
...
+    # check regsitries parameters after voting
+    (
+        rcc_already_spent_amount,
+        rcc_spendable_balanceInPeriod,
+        rcc_period_start_timestamp,
+        rcc_period_end_timestamp
+    ) = rcc_stable_registry.getPeriodState()
+
+    assert rcc_already_spent_amount == 800000000000000000000000
```
Comment (zuzueeka) [inline tests/test_2023_10_31.py:261]: It seems to me better to check the equality of the parameters before the vote
like:
rcc_already_spent_amount_before == rcc_already_spent_amount_after == 800000000000000000000000

File: tests/test_2023_10_31.py [hunk 8]
```diff
@@ -0,0 +1,657 @@
...
+    validate_evmscript_factory_added_event(
+        evs[7],
+        EVMScriptFactoryAdded(
+            factory_addr=atc_stable_topup_factory,
+            permissions=create_permissions(contracts.finance, "newImmediatePayment")
+            + create_permissions(atc_stable_registry, "updateSpentAmount")[2:],
+        ),
+    )
+
+# todo: move to utils
```
Comment (zuzueeka) [inline tests/test_2023_10_31.py:550]: it's already in scripts/utils/test/easy_track_helpers.py

**Comment by bulbozaur** [comment]
Can't see 4th action in script 
iV. Change the on-chain name of node operator with id 27 from 'Prysmatic Labs' to 'Prysm Team at Offchain Labs'

File: tests/test_2023_10_31.py [hunk 9]
```diff
@@ -0,0 +1,632 @@
...
+
+    assert rcc_dai_topup_factory_old in old_factories_list
+    assert pml_dai_topup_factory_old in old_factories_list
+    assert atc_dai_topup_factory_old in old_factories_list
+
+    assert has_payments_permission(contracts.acl, contracts.finance, permission.entity, eth["address"], ldo_holder.address, eth["limit"])
+    assert has_payments_permission(contracts.acl, contracts.finance, permission.entity, steth["address"], ldo_holder.address, steth["limit"])
+    assert has_payments_permission(contracts.acl, contracts.finance, permission.entity, ldo["address"], ldo_holder.address, ldo["limit"])
+    assert has_payments_permission(contracts.acl, contracts.finance, permission.entity, dai["address"], ldo_holder.address, dai["limit"])
+    assert not has_payments_permission(contracts.acl, contracts.finance, permission.entity, usdt["address"], ldo_holder.address, usdt["limit"])
```
Comment (bulbozaur) [inline tests/test_2023_10_31.py:203]: Why are usdt and usdc permissions check with not?

**Comment by kovalgek** [comment]
> Can't find 4th action in script iV. Change the on-chain name of node operator with id 27 from 'Prysmatic Labs' to 'Prysm Team at Offchain Labs'

We decided to skip it for this voting. https://discord.com/channels/761182643269795850/1166003540884455456/1167133297227337779

File: tests/test_2023_10_31.py [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+    dao_balance_after = contracts.lido.balanceOf(AGENT)
+
+    rcc_fund_payout = Payout(token_addr=LIDO, from_addr=contracts.agent, to_addr=rcc_multisig_address, amount=1 * (10**18))
+    pml_fund_payout = Payout(token_addr=LIDO, from_addr=contracts.agent, to_addr=pml_multisig_address, amount=1 * (10**18))
+    atc_fund_payout = Payout(token_addr=LIDO, from_addr=contracts.agent, to_addr=atc_multisig_address, amount=1 * (10**18))
+
+    steth_balance_checker(rcc_multisig_balance_after - rcc_multisig_balance_before, rcc_fund_payout.amount)
+    steth_balance_checker(pml_multisig_balance_after - pml_multisig_balance_before, pml_fund_payout.amount)
+    steth_balance_checker(atc_multisig_balance_after - atc_multisig_balance_before, atc_fund_payout.amount)
+    steth_balance_checker(dao_balance_before - dao_balance_after, 1 * (10**18) + 1 * (10**18) + 1 * (10**18))
```
Comment (zuzueeka) [inline tests/test_2023_10_31.py:86]: why is it an explicit quantity and not the sum of the previous parameters?

File: scripts/vote_2023_10_31.py [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+        ),
+        # 3. Transfer TBA stETH to ATC 0x9B1cebF7616f2BC73b47D226f90b01a7c9F86956
+        make_steth_payout(
+            target_address=atc_multisig_address,
+            steth_in_wei=1 * (10**18),
+            reference="Fund ATC multisig"
+        ),
+        # II. Change the on-chain name of node operator with id 27 from 'Prysmatic Labs' to 'Prysm Team at Offchain Labs'
+        # 4. Grant NodeOperatorsRegistry.MANAGE_NODE_OPERATOR_ROLE to voting
+        encode_permission_grant(
```
Comment (bulbozaur) [inline scripts/vote_2023_10_31.py:80]: Screenshot 2023-10-31 at 14 22 17
Agent already have MANAGE_NODE_OPERATOR_ROLE on Node Operators registry, so you have only to call set name method with agent forward method

**Pull request closed without merging**